Documentation/gettingstarted/dns.rst

```diff
@@ -0,0 +1,203 @@
+
+We will enforce a Cilium network policy which allows:
+
+* Only the ``mediabot`` pods have access to ``api.twitter.com``. 
+* Both the pods have access to the internal ``deathstar`` service.
+
+The following Cilium network policy achieves above access controls. 
+
+.. literalinclude:: ../../examples/minikube/dns-policy.yaml
+
+Let's take a closer look at the policy: 
+
+* The first egress section uses ``toFQDNs: matchName`` specification to allow egress to ``api.twitter.com``. The destination DNS should match exactly the name specified in the rule. The ``endpointSelector`` allows only pods with labels ``class: mediabot`` to have the egress access.
+* The second egress section allows all pods in the ``default`` namespace to access ``kube-dns`` service. Important to note the part ``rules: dns`` that specifies Cilium to inspect and allow DNS lookups matching specified patterns. In this case, inspect and allow all DNS queries. 
+* The third egress section allows all pods in the ``default`` namespace to access the internal ``deathstar`` service. Note that for controlling access to Kubernetes services we use the labels to identify the destination service rather than use DNS. 
+
+Let's apply the policy:
+
+.. parsed-literal::
+
+  $ kubectl create -f \ |SCM_WEB|\/examples/minikube/dns-policy.yaml
+
+
+Testing the policy, we see that ``mediabot`` has access to ``api.twitter.com`` but doesn't have access to any other external service, for e.g., ``help.twitter.com``. And ``spaceship`` pod doesn't have access to twitter. Both the pods still have access to the internal ``deathstar`` service. 
+
+.. parsed-literal::
+
+    $ kubectl exec -it mediabot -- curl -sL https://api.twitter.com
+    ...
+    ...
+
+    $ kubectl exec -it mediabot -- curl -sL https://help.twitter.com
+    ^C 
+
+    $ kubectl exec -it spaceship -- curl -sL https://api.twitter.com
+    ^C
+
+    $ kubectl exec -it spaceship -- curl -sL deathstar.default.svc.cluster.local/v1
+    {
+	"name": "Death Star",
+        ...
+        ...
+    }
+
+    $ kubectl exec -it mediabot -- curl -sL deathstar.default.svc.cluster.local/v1
+    {   
+        ...
+        ...
+    }
+
+
+Step 6: DNS Policies Using Patterns 
+===================================
+
+The above policy controlled DNS access based on exact match of the DNS domain name. Often it is required to allow access to a subset of domains. A common example is to control access for cloud services such as AWS S3 which have buckets and regions encoded in the DNS domains. For e.g., the S3 bucket URLs can be of the form, ``<bucketname>.s3.<region>.amazonaws.com``. A pattern based rule such as ``*.s3.*.amazonaws.com`` is needed to allow access to all the S3 buckets irrespective of the region.
+
+Let's use the above example and allow ``mediabot`` pods to access any twitter sub-domain i.e. allow a pattern ``*.twitter.com``. We will achieve this by changing the ``toFQDN`` rule to use ``matchPattern`` instead of ``matchName``.
```

Comment:
Why use S3 for the description but then not use it for the example? It would be a lot more concrete if the example was directly usable.

Reply:
I was not sure what public S3 bucket to use. Didn't want to use the covalent assets bucket. The S3 served as additional example for pattern based policies.

Reply:
Removed the S3 parts. It was meant to serve as another example but the link to DNS pattern section in docs serves that purpose.